utilities/checkpoint/checkpoint_impl.cc

```diff
@@ -180,10 +200,15 @@ Status CheckpointImpl::CreateCheckpoint(const std::string& checkpoint_dir,
     ROCKS_LOG_INFO(db_options.info_log, "Snapshot sequence number: %" PRIu64,
                    sequence_number);
   } else {
-    // clean all the files we might have created
     ROCKS_LOG_INFO(db_options.info_log, "Snapshot failed -- %s",
                    s.ToString().c_str());
-    CleanStagingDirectory(full_private_path, db_options.info_log.get());
+    // clean all the files and directory we might have created
+    Status del_s =
+        CleanStagingDirectory(full_private_path, db_options.info_log.get());
```

Comment:
Should we propagate any errors from this call too (for the same fault injection related reasons as in the PR description)?

Reply:
In this case, s is already not-okay and s should take higher precedence than del_s for consideration of stress test.  If del_s is an injected error, s is either an injected error (get treated just as the same as we propagate del_s) or a different error (needs to be crashed). The failure to clean up full_private_path here will be handled here https://github.com/facebook/rocksdb/pull/12894/files#diff-9e650226991dd4e7b9e009c58803a3d07c099c058ab04739df41f8cec62cfdfbR126next time